src/frontend/src/lib/utils/transactions.utils.ts

```diff
@@ -0,0 +1,49 @@
+import BtcTransaction from '$btc/components/transactions/BtcTransaction.svelte';
+import type { BtcTransactionUi } from '$btc/types/btc';
+import type { CertifiedStoreData } from '$lib/stores/certified.store';
+import type { TransactionsData } from '$lib/stores/transactions.store';
+import type { Token } from '$lib/types/token';
+import type { AllTransactionsUi } from '$lib/types/transaction';
+import {
+	isNetworkIdBTCMainnet,
+	isNetworkIdEthereum,
+	isNetworkIdICP
+} from '$lib/utils/network.utils';
+import { nonNullish } from '@dfinity/utils';
+
+/**
+ * Maps the transactions stores to a unified list of transactions with their respective components.
+ *
+ * @param tokens - The tokens to map the transactions for.
+ * @param $btcTransactions - The BTC transactions store.
+ */
+export const mapAllTransactionsUi = ({
+	tokens,
+	$btcTransactions
+}: {
+	tokens: Token[];
+	$btcTransactions: CertifiedStoreData<TransactionsData<BtcTransactionUi>>;
+}): AllTransactionsUi =>
+	tokens.reduce<AllTransactionsUi>((acc, { id: tokenId, network: { id: networkId } }) => {
+		if (isNetworkIdBTCMainnet(networkId) && nonNullish($btcTransactions)) {
```

Comment:
Maybe remove the `nonNullish($btcTransactions)` and handle the case within the if? That way the loop won't proceed with the other ifs.

Reply:
yes

Reply:
done!